### ISSUE TITLE
Double-digit thread support

### ISSUE THREAD

**Issue by xxxserxxx** [xxxserxxx/gotop#40]
i don't believe there's an issue related to this, but definitely close and link if there is. 😅 

i have a TR1920X, which has 24-threads. running `gotop -p`, will list all the threads, however, i typically can't see more than the first 12 or so unless i fully maximize the window.

**would it be possible to adaptively place the cpus?* (meaning, if the window which lists the cpus is able to "detect" an overflow, it repaints as a column?)

i know that i would take a temporal update hit to have that – meaning if it reduces the update frequency, that's definitely an acceptable trade-off, imo. (or maybe we can pass a flag?)

Original ticket https://github.com/cjbassi/gotop/issues/158
Original submitter @ionlights

**Comment by ionlights** [comment]
@xxxserxxx are you planning to work on this? 🤔 (just got a notif, so i was curious.)

**Comment by xxxserxxx** [comment]
@ionlights Yes, I am.  Here's a bit of copypasta, since this question is (understandably) coming up frequently:

Caleb is not maintaining gotop any longer, but also doesn't want anybody else maintaining his repository, nor is he comfortable with moving issues. I'm continuing development and maintenance of gotop, so I've forked and copied the tickets so that I can keep working on them.

See [this issue](https://github.com/cjbassi/gotop/issues/187) for more information.

**Comment by xxxserxxx** [comment]
@ionlights there have been [new releases](https://github.com/xxxserxxx/gotop/releases), as well.

**Comment by xxxserxxx** [comment]
@ionlights I think I understand what you mean.  You want to see a layout like this:
```
 +- CPU Usage -----------------------+
 |                                   |
 | CPU0 10%   CPU5 10%               |
 | CPU1  5%   CPU6  5%               |
 | CPU2 10%   CPU7 10%               |
 | CPU3  5%                          |
 | CPU4 10%                          |
 +-----------------------------------+
```
Or were you thinking of something different?

In the meantime, can you check out a recent release and see if the ability to define your own layouts helps?  If you download a binary, [download this file](https://raw.githubusercontent.com/xxxserxxx/gotop/master/layouts/procs) and save it to `pwd` directory, or copy it to `$XDG_CONFIG_HOME/gotop/`, and then run `gotop -l procs`.  The layout file has to be either in the same directory you're in, or in the `gotop` config directory.

The `layouts/` directory has more examples, and the README explains the syntax.  With this, you can make the procs widget as tall as the console, and maybe see all of your CPUs.  It's not a full solution, because Ryzen ThreadRipper exists.  But maybe it'll be a sufficient work-around until this ticket is closed

**Comment by ionlights** [comment]
So, that layout is correct. I think something that might be useful (though probably harder) would be to adapt based on the screen-size. e.g. 
```
+- CPU Usage -------------------------+
|                                     |
| CPU00 10%   CPU03  5%   CPU06 10%   |
| CPU01  5%   CPU04 10%   CPU07  5%   |
| CPU02 10%   CPU05 10%   CPU08 10%   |
|                                     |
+-------------------------------------+
```
Though I'd probably classify this as a "stretch" implementation.

I'll also give your proposed interim solution a look in a few minutes as well.

Thanks for doing this! :smiley:

**Comment by ionlights** [comment]
<img width="1795" alt="Screen Shot 2020-02-25 at 3 25 28 PM" src="https://user-images.githubusercontent.com/5000729/75284560-103f5480-57e3-11ea-9b92-66fd9443dec4.png">

I was finally able to update to 3.3.1 on Arch Linux (`linux-lts`), with kernel `5.4.21`. (Need to use `linux-lts` because of `zfs`,(

**Comment by ionlights** [comment]
So, I'm not sure that this is supported? This machine has 24-threads. Screenshots both in regular window and `tmux`.

<img width="1527" alt="Screen Shot 2020-05-03 at 16 21 08" src="https://user-images.githubusercontent.com/5000729/80924949-32f96680-8d5a-11ea-9f0f-74f20e30e3dc.png">
<img width="1527" alt="Screen Shot 2020-05-03 at 16 20 52" src="https://user-images.githubusercontent.com/5000729/80924952-35f45700-8d5a-11ea-9ca4-a7470ef7311c.png">

**Comment by xxxserxxx** [comment]
@ionlights hi John.  This isn't in the 3.5 branch -- the one you're using for NVidia.  I try to keep the "Milestone" field (on the left here in Github) accurate; as you can see, this was fixed in the 4.0 branch, which hasn't yet been released.  If you try that branch and it still doesn't work, do let me know.

Sorry for the confusion.

**Comment by ionlights** [comment]
Ahh. Okay. Wasn’t sure since I saw it closed. Will checkout v4 in a few days and lyk if anything breaks. 🙂
